File: libs/initialize/initialize.ts
//% color=#800000 weight=100 icon="\uf019" block="Initialise"
namespace initialize {


    //% blockId=initialize_initUnit block="init raspberry pi to| %name |as id for %t" weight=94
    //% shim=initialize::initUnit
    export function initUnit(id:string, t: InitType) {
    }

    //% blockId=initialize_resetUnit block="clear initialisation" weight=93
    //% shim=initialize::resetUnit
    export function resetUnit() {
    }
}

File: libs/initialize/initialize.cpp
#include "pxt.h"
#include "InitService.h"

enum class InitType{
   school = 0,
    raspberry = 1,
};

/**
* share
* use
*/

InitService InitService(uBit.radio);

//% color=#51008B weight=100 icon="\uf019" block="Initialise"
namespace initialize {

    int init() {
        return enableRadio();
    }

    //%
    void initUnit(StringData* initData, InitType t)
    {
        init();

        ManagedString result;
        if(t == InitType::school)
        {
            result = InitService.setSchoolId(ManagedString(initData));
        }
        else
        {
            result = InitService.setPiId(ManagedString(initData));
        }
        uBit.display.scroll(result);
    }
};
